### PR TITLE
Add navigator

### PR DESCRIPTION
**Pull request by jasalisbury**
**Description of change**

The navigator allows us to split up large forms into multiple pages. It has a side navigation that allows any page to be selected. The side navigation displays the state of each page and is stickied to the top of the page. Some notes:

 * Adds a new library `react-helmet` to help update page titles. This library helps keep the page title updated with the page the user is viewing in the activity report.
 * Reduced the use of `act(...)` in tests. Previously we would get warnings that looked like `Warning: An update to *COMPONENT* inside a test was not wrapped in act(...).`. This indicates updates to the DOM were happening outside of a test. I've updated tests to not need to be wrapped in act. Some tests still use act but they deal with third party libraries or using mock timers.
 * Started replacing `fireEvent` with `userEvent` per this [suggestion](https://testing-library.com/docs/dom-testing-library/api-events/). Will make a separate PR to switch everything over
 * The activity summary page is pretty long, it would not surprise me if it gets broken down into multiple pages at some point.
 * ~After https://github.com/adhocteam/Head-Start-TTADP/pull/62 is merged in I'll update this branch to deploy to sandbox (unless you are using sandbox @SarahJaine!).~

**How to test**

1) Browse to https://tta-smarthub-sandbox.app.cloud.gov/activity-reports and fill out the form

OR

1) Pull down branch
2) Install deps
3) Start dev environment, login and browse to `localhost:3000/activity-report`
4) Fill out the form

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/140

**Checklist**

- [x] Meets issue criteria
- [x] Code tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [n/a] Documentation updated
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified

### CODE REVIEW

File: frontend/src/components/Navigator/components/SideNav.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
+/*
+  Stickied left nav for the navigator. Should not have any state, just displays
+  the nav items passed in as props. This component has lots of custom styles
+  defined. Note the nav is no longer sticked once we hit mobile widths (640px)
```
Comment (kryswisnaskas) [inline frontend/src/components/Navigator/components/SideNav.js:4]: > sticked

 typo?

File: frontend/src/components/Navigator/index.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
+/*
+  The navigator is a component used to show multiple form pages. It displays a stickied nav window
+  on the left hand side with each page of the form listed. Clicking on an item in the nav list will
+  display that item in the content section. The navigator keeps track of the "state" of each page.
+  In the future logic will be added to the navigator to prevent the complete form from being
+  submitted until every page is completed.
+*/
```
Comment (kryswisnaskas) [inline frontend/src/components/Navigator/index.js:7]: 🎉

**Review by kryswisnaskas — COMMENTED** [review]
Very nice!

A lot of great features.

I did notice the UI jump a bit when navigating sections, but this is minor.

I wonder if we should start adding validations to the UI. I know we have a separate PR for validations. Do you think it would be easier to handle them all at once or start implementing them at least partially, e.g. the file uploads validations?

We should probably also add some cucumber tests, either here or in a different PR.

**Comment by jasalisbury** [comment]
> I wonder if we should start adding validations to the UI. I know we have a separate PR for validations. Do you think it would be easier to handle them all at once or start implementing them at least partially, e.g. the file uploads validations?

There shouldn't be too many fields that require UI validations as most fields are select-boxes or radio/check-boxes. The fields that do require validations (text or number fields) should be easy to use because we are using `react-hook-form`. I would suggest we add the missing validations in one PR and whenever we add additional fields to the form in the future we add the validation as well.

> We should probably also add some cucumber tests, either here or in a different PR.

My vote would be a different PR, this PR is already pretty big.

File: frontend/src/components/Navigator/components/SideNav.js (hunk 1 above)
Comment (jasalisbury) [reply frontend/src/components/Navigator/components/SideNav.js:4]: Yep, thanks!

**Review by kryswisnaskas — APPROVED** [review]

**Pull request merged**